### ISSUE TITLE
query() is now unusable?

### ISSUE THREAD

**Issue by Holt59** [typst/typst#772]
After updating to latest `typst`, I get the following error when using `query`:

```
error: missing argument: location
    ┌─ doc\_00-template.typ:226:24
    │
226 │     let elements = query(heading, after: loc)
    │          
```

According to the documentation for the `location` parameter:

> Only one of this, before, and after shall be given.

So, `location` is now mandatory but it should not be given together with `before` or `after`? So `before` and `after` are unusable? What is the correct new usage of `query` to match the line above?

**Comment by laurmaedje** [comment]
The documentation is not yet fully up to date. You should write:
`let elements = query(selector(heading).after(loc), loc)`

It's a bit duplicate with the location in this case, but its a lot more flexible than before, because you can arbitrarily combine `before` and `after` restrictions with other selectors.

**Comment by Holt59** [comment]
> 

I changed to simply `query(heading, loc)` which is equivalent here I guess? Seems to work for now, but thanks for the clarification.

**Comment by leo848** [comment]
@Holt59 No, it is not equivalent. `query(heading, loc)` will not query after the current location, but the entire document including the current context. If you e.g. use it in a table of contents (TOC), the TOC itself will be contained in the TOC.

**Comment by Holt59** [comment]
> @Holt59 No, it is not equivalent. `query(heading, loc)` will not query after the current location, but the entire document including the current context. If you e.g. use it in a table of contents (TOC), the TOC itself will be contained in the TOC.

Thanks for the clarification, I'm not using a `heading` for my TOC so that's ok for now but good to know for future documents.
